### ISSUE TITLE
Wrong warning on useless_vec

### ISSUE THREAD

**Issue by BusyJay** [Manishearth/rust-clippy#1095]
Consider following code:
```
struct A;

impl A {
    fn move_self(self) {
        // do something
    }
}

for a in vec![A, A, A] {
    a.move_self()
}
```
Because a need to be moved, so here using a Vector. But clippy will complain:

> ```
warning: useless use of `vec!`, #[warn(useless_vec)] on by default
      for a in vec![A, A, A] {
               ^~~~~~~~~~~~~
help: you can use a slice directly
      for a in &[A, A, A] {
help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#useless_vec
```
